reference-implementation/test/readable-stream.js

```diff
@@ -290,6 +289,96 @@ test('ReadableStream is able to pull data repeatedly if it\'s available synchron
   });
 });
 
+test('ReadableStream does not call pull until previous pull\'s promise fulfills', t => {
+  var resolve;
+  var returnedPromise;
+  var timesCalled = 0;
+  var rs = new ReadableStream({
+    pull(enqueue) {
+      ++timesCalled;
+      enqueue(timesCalled);
+      returnedPromise = new Promise(r => { resolve = r; });
+      return returnedPromise;
+    }
+  });
+
+  t.equal(rs.state, 'waiting', 'stream starts out waiting');
+
+  rs.ready.then(() => {
+    t.equal(rs.state, 'readable', 'stream becomes readable (even before promise fulfills)');
+    t.equal(timesCalled, 1, 'pull is not yet called a second time');
+    t.equal(rs.read(), 1, 'read() returns enqueued value');
+
+    setTimeout(() => {
+      t.equal(timesCalled, 1, 'after 30 ms, pull has still only been called once');
+
+      resolve();
+
+      returnedPromise.then(() => {
```

Comment:
I agree that Promise.resolve().then(...) would work. But don't you think a good way to verify "pull() will be called once the returnedPromise is resolved" is `returnedPromise.then(() => checkThatPullWasCalled())`?

One way of looking at it is that I could move the resolve() call on L315 somewhere else randomly (say, in a top-level `setTimeout(resolve, 100)`). L317 is meant to stand on its own independent of L315.